pkg/operator/vspherecontroller/vspherecontroller.go

```diff
@@ -626,23 +635,25 @@ func (c *VSphereController) createCSISecret(
 		return fmt.Errorf("cloud config %s/%s does not contain key %q", cloudConfigNamespace, cloudConfig.Name, cloudConfig.Key)
 	}
 
-	var cfg vsphere.VSphereConfig
-	err = gcfg.ReadStringInto(&cfg, cfgString)
+	cfg := vclib.VSphereConfig{}
+	err = cfg.LoadConfig(cfgString)
 	if err != nil {
-		return err
+		return fmt.Errorf("unable to fetch cloud provider config: %v", err)
 	}
 
-	storageApiClient := storageclasscontroller.NewStoragePolicyAPI(ctx, c.vSphereConnection, infra)
+	// Pass in first vcenter for now.  I think this logic is no longer valid, but need to confirm if we are wanting
+	// multi vcenter to support storage migration.
+	storageApiClient := storageclasscontroller.NewStoragePolicyAPI(ctx, c.vSphereConnections[0], infra)
```

Comment:
So, I do not very much like just randomly picking first `vSphereConnection` as default one. This is what we were discussing - we will have to probably degrade clusters if more than one VC exists and migration needs to be supported. cc @rvanderp3

Reply:
I've changed this to be related to each vCenter.  We talked about how this is used for migration and that migration will be disabled in multi vCenter.  Also, in the case of multiple failure domains in the same vCenter, the first FD is usually (or should be) the original FD so that we can make sure to migrate those old volumes.  It is also possible for this to only look at the old ini config to get the default settings.